### ISSUE TITLE
Room list not always received

### ISSUE THREAD

**Issue by JulesThuillier** [cxz456999/NATP#2]
Hello again !

Here is my setup : I have 2 instances of the game on the same computer.
On the first instance, I start a server or host. The connection and room information are received in the Signaling server.
On the second instance I try to get the Room list (client not yet started). I see the GetRoomListRequest in the Signaling server console, but I do not always receive the response. Usually I have to exit and restart my second instance to get the Room List properly (no need to restart the server).

It seems to happen only when Issue #1 happens. Or when I try to start stop the server multiple times.

![image](https://user-images.githubusercontent.com/8751523/84522208-36aad200-aca4-11ea-8efd-9c8baafe9b54.png)

**Comment by cxz456999** [comment]
Hi, thx for reporting bugs.
I'm trying to duplicate the bug you said, but I failed.
I don't think that is the reason, but maybe it is, here is how it works
1. client(host) connect to signaling server.
2. after connected, connected event trigger on both client and signaling server.
3. client send create/get room list request to signaling server.
4. signaling server response
give me some time to find the reason...

**Comment by cxz456999** [comment]
Is there high latency between client/host and signaling server?

**Comment by JulesThuillier** [comment]
Hi ! Thanks for the answer.

Concerning the latency, I am using a distant server, the ping is 16ms, so not much latency here.

For your informations, I have seen that NATPClient.OnAllocateResponseEvent is not always executed.  But that is more related to issue #1.

Please note those issue mostly happens when starting stopping server multiple times in row. So maybe it has something to do with timeout / disconnection. I'll let you know if I find more info.

**Comment by cxz456999** [comment]
Have you seen the logs from the coturn server? 
I think maybe there are something wrong to the coturn sever so that OnAllocateResponseEvent didn't execute.

**Comment by JulesThuillier** [comment]
Yes I think that sometimes the Coturn connection does not re-establish because it did not timeout, so it thinks it is still connected. I would need to investigate more, I'll check that next week ;)

**Comment by cxz456999** [comment]
OH! that is the point!
You can check if Coturn shows this kind of messages
"handle_udp_packet: New UDP endpoint: ...."
means it is a new connection for the Coturn, otherwise I think there will be some error messages like
"error 437: Mismatched allocation: wrong transaction ID"
Anyway, thanks for interesting in this project! I'm trying to make the other version that without Coturn server, I think the new version will be easier to setup and use.

**Comment by cxz456999** [comment]
NATPTransport.unitypackage updated.
Update netcoreserver and fix some code that I think cause the bugs you said.

**Comment by JulesThuillier** [comment]
Sweet ! I'm gonna try that ! 
Thanks for the support !
A version without coturn would definitely make it easier to deploy, but it's alright with your doc ;)

**Comment by JulesThuillier** [comment]
I think both issues are solved !!!

**Comment by cxz456999** [comment]
WOW!! Nice work! 
This is new version, you can try it if you want~
https://github.com/cxz456999/NATPTransport
